test/tests/functional/pbs_cray_vnode_pool.py

```
@@ -0,0 +1,325 @@
+        """
+        Invalid vnode_pool values shall result in errors.
+        """
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr_A = {'vnode_pool': '-1'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': '0'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': 'a'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+    def test_two_moms_single_vnode_pool(self):
+        """
+        Same vnode_pool for two moms shall result in one mom being the
+        inventory mom and the other the non-inventory mom.
+        The inventory mom goes down (e.g. killed).
+        Compute nodes remain up even when the inventory mom is killed,
+        since another mom is reporting them.
+        Check that a new inventory mom is listed in the log.
+        Bring up killed mom.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {"log_events": -1})
+
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
```

The test requires two moms and if it were to silently skip the test then we would have no failure or error to indicate that the test used less than two moms.